beacon-light-client/plonky2/input_fetchers/balance_verification/deposits_accumulator/lib/scheduler.ts

```diff
@@ -0,0 +1,161 @@
+import { Redis } from '@dendreth/relay/implementations/redis';
+import { KeyPrefix, WorkQueue } from '@mevitae/redis-work-queue';
+import { ethers } from 'ethers';
+import { getEvents } from './event-fetcher';
+import ValidatorsAccumulator from '../../../../../solidity/artifacts/contracts/validators_accumulator/ValidatorsAccumulator.sol/ValidatorsAccumulator.json';
+import chalk from 'chalk';
+
+enum Events {
+  Deposited = 'Deposited',
+}
+const EVENTS_BATCH_SIZE = 10_000;
+
+export abstract class Scheduler {
+  protected redis: Redis;
+  protected queue: any;
+  protected provider: ethers.providers.JsonRpcProvider;
+  protected contract: ethers.Contract;
+  protected syncBlock: number;
+  protected ssz: any;
+
+  async init(
+    options: any,
+    constants: {
+      queue: string;
+      latestBlock: string;
+    },
+  ): Promise<void> {
+    this.redis = new Redis(options['redis-host'], options['redis-port']);
+    this.queue = new WorkQueue(new KeyPrefix(`${constants.queue}`));
+    this.provider = new ethers.providers.JsonRpcProvider(options['rpc-url']);
+    this.contract = new ethers.Contract(
+      options['address'],
+      ValidatorsAccumulator.abi,
+      this.provider,
+    );
+
+    let latestLoggedBlock = await this.redis.get(constants.latestBlock);
+    if (!latestLoggedBlock) {
+      latestLoggedBlock = (
+        (await this.provider.getBlockNumber()) - EVENTS_BATCH_SIZE
+      ).toString();
+    }
```

Comment:
We need to pass in the block number the contract was deployed in

Reply:
Upon starting the script, you can pass it using the --sync-block option.

Reply:
Maybe you should set `latestLoggedBlock` when you dump the events

Reply:
Ah, so it's used for rerunning the script. Tbh I would rather crash if latestLoggedBlock is unset and sync-block is not supplied. The identifier  latestLoggedBlock is a little misleading since it's really the oldest unlogged block